tests/MongoDB.Driver.Tests/Linq/Translators/PredicateTranslatorValidationTests.cs

```diff
@@ -0,0 +1,161 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Driver.Linq;
+using Xunit;
+
+namespace MongoDB.Driver.Tests.Linq.Translators
+{
+    public class PredicateTranslatorValidationTests
```

Comment:
I decided to create(move from the previous PR) a new test file since these tests work in a little bit another way than `PredicateTranslatorTests`. Here, we want to check throwing (or not) exception for different cases and generated an error message. Also, it's more simple to have a separate `TestObject` class than usual `Root`. But I may move these tests to the `PredicateTranslatorTests` if it is needed.